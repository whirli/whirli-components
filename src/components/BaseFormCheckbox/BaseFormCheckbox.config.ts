// @ts-ignore
import { PropType } from '@composition';
import { ComponentStyles, ComponentProps } from '@whirli-components/@types/components';

// Local config
import config from '@whirli-local/components/BaseFormCheckbox/BaseFormCheckbox.config';

// Package config
import { PropKeys } from './BaseFormCheckbox.constants';

// Shared Config
import { SharedConfigProps } from '@whirli-components/components/BaseFormInput/BaseFormInput.config';

export const ConfigStyles: ComponentStyles = {
  ...config.styles,
};

export const ConfigProps: ComponentProps = {
  [PropKeys.MODEL_VALUE]: {
    required: true,
    type: Boolean as PropType<boolean>,
  },
  ...SharedConfigProps,
  ...config.props,
};
