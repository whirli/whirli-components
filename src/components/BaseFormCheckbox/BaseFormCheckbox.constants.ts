export const PropKeys: Record<string, string> = {
  VALUE: 'value',
};

export const PropValues: Record<string, string[]> = {};
