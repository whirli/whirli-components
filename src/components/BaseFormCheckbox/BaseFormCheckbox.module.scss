@import '@whirli-components/styles/Breakpoints.module.scss';

// default styles

// breakpoint styles
$package-breakpoint-styles: (

);
