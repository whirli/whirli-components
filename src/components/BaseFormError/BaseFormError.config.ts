// @ts-ignore
import { ComponentStyles, ComponentProps } from '@whirli-components/@types/components';

// Local config
import config from '@whirli-local/components/BaseFormError/BaseFormError.config';

// Package config
export const ConfigStyles: ComponentStyles = {
  ...config.styles,
};

export const ConfigProps: ComponentProps = {
  ...config.props,
};
