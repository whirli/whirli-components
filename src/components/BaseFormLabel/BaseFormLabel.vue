<template>
  <label :for="props.name" :class="[classes, requiredClass]"><slot /></label>
</template>

<script setup lang="ts">
// Styles
// @ts-ignore
import styles from '@whirli-local/components/BaseFormLabel/BaseFormLabel.module.scss?module';

// Data
import { ConfigStyles, ConfigProps } from './BaseFormLabel.config';

// Types
import { ComponentStyles as ComponentStylesInterface } from '@whirli-components/@types/components';
import { Props } from '@whirli-components/@types/props';

const ComponentStyles: ComponentStylesInterface = ConfigStyles;

const props: Props = defineProps(ConfigProps);

// Classes
import useClasses from '@whirli-components/@use/class';
const { makeClasses } = useClasses();
const classes: string[] = [styles.label, ...makeClasses(ComponentStyles, props, styles)];
const requiredClass: string = props.required ? styles.required : '';
</script>
