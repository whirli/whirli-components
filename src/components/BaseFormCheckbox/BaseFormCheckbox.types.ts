import { PropKeys, PropValues } from './BaseFormCheckbox.constants';
