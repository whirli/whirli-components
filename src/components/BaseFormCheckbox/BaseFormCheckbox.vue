<template>
  <BaseFormGroupLayoutSecondary v-bind="$props">
    <input
      type="checkbox"
      :id="props.name"
      :name="props.name"
      :class="classes"
      :disabled="props.disabled"
      :aria-disabled="props.disabled"
      :aria-describedby="getErrorId(props.name, $props.error)"
      :checked="props.modelValue"
      @change="updateValue($event)"
    />
  </BaseFormGroupLayoutSecondary>
</template>

<script setup lang="ts">
// Vue
import { defineEmits } from '@composition';

// Styles
// @ts-ignore
import sharedStyles from '@whirli-local/components/BaseFormInput/BaseFormInput.module.scss?module';
// @ts-ignore
import styles from '@whirli-local/components/BaseFormCheckbox/BaseFormCheckbox.module.scss?module';

// Data
import { ConfigStyles, ConfigProps } from './BaseFormCheckbox.config';

// Types
import { ComponentStyles as ComponentStylesInterface } from '@whirli-components/@types/components';
import { Props } from '@whirli-components/@types/props';

// Components
import BaseFormGroupLayoutSecondary from '@whirli-components/components/BaseFormGroup/BaseFormGroupLayoutSecondary.vue';

// Composables
import useClasses from '@whirli-components/@use/class';
import useForm from '@whirli-components/@use/form';

// @ts-ignore
const emit = defineEmits<{
  (event: 'update:modelValue', value: boolean): void;
  (event: 'reset:errors'): void;
}>();

const ComponentStyles: ComponentStylesInterface = ConfigStyles;

// @ts-ignore
const props: Props = defineProps(ConfigProps);

// Classes
const { makeClasses } = useClasses();
const classes = [
  sharedStyles['input-shared'],
  styles.checkbox,
  ...makeClasses(ComponentStyles, ConfigProps, props, styles),
  ...makeClasses(ComponentStyles, ConfigProps, props, sharedStyles),
];

const updateValue: (event: Event) => void = (event: Event) => {
  emit('update:modelValue', Boolean((event.target as HTMLInputElement).checked));
  emit('reset:errors');
};

const { getErrorId } = useForm();
</script>
