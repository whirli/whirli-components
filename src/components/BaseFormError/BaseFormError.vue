<template>
  <BaseText v-bind="$attrs" :class="classes" v-if="$slots.default"><slot /></BaseText>
</template>

<script setup lang="ts">
// Styles
// @ts-ignore
import styles from '@whirli-local/components/BaseFormError/BaseFormError.module.scss?module';

// Data
import { ConfigStyles, ConfigProps } from './BaseFormError.config';

// Types
import { ComponentStyles as ComponentStylesInterface } from '@whirli-components/@types/components';
import { Props } from '@whirli-components/@types/props';

const ComponentStyles: ComponentStylesInterface = ConfigStyles;

// @ts-ignore
const props: Props = defineProps(ConfigProps);

// Classes
import useClasses from '@whirli-components/@use/class';
import BaseText from '@whirli-components/components/BaseText/BaseText.vue';
const { makeClasses } = useClasses();
const classes = [styles.error, ...makeClasses(ComponentStyles, ConfigProps, props, styles)];
</script>
