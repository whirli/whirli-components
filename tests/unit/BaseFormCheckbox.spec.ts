import { mount } from '@vue/test-utils';
import BaseFormCheckbox from '@/components/BaseFormCheckbox/BaseFormCheckbox.vue';
import BaseFormLabel from '@/components/BaseFormLabel/BaseFormLabel.vue';
import BaseFormError from '@/components/BaseFormError/BaseFormError.vue';

let wrapper: any;
const label = 'Hello World';
const error = 'Hello Error';
const name = 'input-name';

describe('BaseFormCheckbox.vue', () => {
  beforeEach(() => {
    wrapper = mount(BaseFormCheckbox, {
      props: {
        label,
        error,
        name,
      },
    });
    wrapper.vm.updateValue = jest.fn();
  });

  it('renders the component', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('displays the correct label text', () => {
    expect(wrapper.findComponent(BaseFormLabel).text()).toMatch(label);
  });

  it('displays the correct error text', () => {
    expect(wrapper.findComponent(BaseFormError).text()).toMatch(error);
  });

  it('input has the correct name', () => {
    expect(wrapper.find('input').element.getAttribute('name')).toBe(name);
  });

  it('updates value on change', async () => {
    wrapper.find('input').trigger('change');
    await wrapper.vm.$nextTick();
    expect(wrapper.vm.updateValue).toHaveBeenCalled();
  });
});
