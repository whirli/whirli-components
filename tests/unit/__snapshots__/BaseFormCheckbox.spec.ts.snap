// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseFormCheckbox.vue renders the component 1`] = `
<div
  class=""
  disabled="false"
  modelvalue="false"
>
  
  <div
    class=""
  >
    
    <div
      class=""
    >
      
      
      <label
        class=""
        for="input-name"
      >
        
        Hello World
        
      </label>
      
      <input
        aria-disabled="false"
        class=""
        id="input-name"
        name="input-name"
        type="checkbox"
      />
      
      
      
    </div>
    <div
      class=""
    >
      
      
      <span
        class=""
        id="input-name-error"
      >
        
        
        Hello Error
        
        
      </span>
      
      
    </div>
    
  </div>
  
</div>
`;
